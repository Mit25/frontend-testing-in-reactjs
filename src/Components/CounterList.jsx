import React from 'react';
import Counter from "./Counter";

const createRange = num => Array.from(Array(num).keys());

export default class CounterList extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            noOfCounters: 1
        }
    }

    render() {
        const counters = createRange(this.state.noOfCounters);

        return (
            <div>
                <button id='btn-add' onClick={e => this.addCounter(e)}>Add Counter</button>
                <button id='btn-reset' onClick={e => this.reset(e)}>Reset Counter List</button>
                <ul>
                    {counters.map(num => this.renderCounter(num))}
                </ul>
            </div>
        )
    }

    addCounter(event) {
        this.setState({
            noOfCounters: this.state.noOfCounters + 1
        });
    }

    reset(event) {
        this.setState({
            noOfCounters: 1
        });
    }

    renderCounter(index){
        return (
            <li key={index}>
                <Counter />
            </li>
        )
    }

}

CounterList.defaultProps = {

};