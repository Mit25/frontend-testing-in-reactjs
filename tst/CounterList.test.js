import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import CounterList from '../src/Components/CounterList';

describe('Counter component', () => {

    it('snapshot testing for Counter component', () => {
        const tree = renderer.create(<CounterList />).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render CounterList component with initial one counter', () => {
        const wrapper = shallow(<CounterList />);

        expect(wrapper.find('Counter').length).toBe(1);
    });

    it('should add Counter while clicking on Add button', () => {
        const wrapper = shallow(<CounterList />);

        wrapper.find('#btn-add').simulate('click');
        expect(wrapper.find('Counter').length).toBe(2);
    });

    it('should reset Counter while clicking on Reset button', () => {
        const wrapper = shallow(<CounterList />);

        wrapper.find('#btn-add').simulate('click');
        expect(wrapper.find('Counter').length).toBe(2);

        wrapper.find('#btn-reset').simulate('click');
        expect(wrapper.find('Counter').length).toBe(1);
    });
});