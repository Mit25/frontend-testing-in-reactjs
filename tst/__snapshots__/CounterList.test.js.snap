// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component snapshot testing for Counter component 1`] = `
<div>
  <button
    id="btn-add"
    onClick={[Function]}
  >
    Add Counter
  </button>
  <button
    id="btn-reset"
    onClick={[Function]}
  >
    Reset Counter List
  </button>
  <ul>
    <li>
      <div>
        <div>
          <span
            id="spn-counter-text"
          >
            The counter: 
          </span>
          <span
            id="spn-count"
          >
            0
          </span>
        </div>
        <button
          id="btn-increment"
          onClick={[Function]}
        >
          Increment
        </button>
        <button
          id="btn-decrement"
          onClick={[Function]}
        >
          Decrement
        </button>
        <button
          id="btn-reset"
          onClick={[Function]}
        >
          Reset
        </button>
      </div>
    </li>
  </ul>
</div>
`;
